From:        Martin Schwidefsky <schwidefsky@de.ibm.com>
Subject:     Correct init-Sequence
Description: At the time the Timer are initialised, the page_discard
             needs to be initialised

Acked-by:    Ihno Krumreich <Ihno@suse.de>

--- linux-2.6.27/arch/s390/kernel/time.c	2008-10-16 20:04:06.528170726 +0200
+++ linux-2.6.27/arch/s390/kernel/time.c.new	2008-10-16 20:05:14.108140559 +0200
@@ -258,16 +258,16 @@ void __init time_init(void)
 					      &ext_int_etr_cc) != 0)
 		panic("Couldn't request external interrupt 0x1406");
 
+#ifdef CONFIG_PAGE_STATES
+	page_discard_init();
+#endif
+
 	/* Enable TOD clock interrupts on the boot cpu. */
 	init_cpu_timer();
 
 #ifdef CONFIG_VIRT_TIMER
 	vtime_init();
 #endif
-
-#ifdef CONFIG_PAGE_STATES
-	page_discard_init();
-#endif
 }
 
 /*
